src/app/tests/suites/commands/interaction_model/InteractionModel.cpp

```diff
@@ -0,0 +1,174 @@
+    params.mEventPathParamsListSize     = 1;
+    params.mpAttributePathParamsList    = nullptr;
+    params.mAttributePathParamsListSize = 0;
+
+    // TODO: Add data version supports
+    mReadClient =
+        std::make_unique<chip::app::ReadClient>(chip::app::InteractionModelEngine::GetInstance(), device->GetExchangeManager(),
+                                                mBufferedReadAdapter, chip::app::ReadClient::InteractionType::Subscribe);
+    VerifyOrReturnError(mReadClient != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    return mReadClient->SendRequest(params);
+}
+
+CHIP_ERROR InteractionModel::SubscribeAttribute(const char * identity, chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                chip::AttributeId attributeId, uint16_t minInterval, uint16_t maxInterval,
+                                                bool fabricFiltered)
+{
+    chip::DeviceProxy * device = GetDevice(identity);
+    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    chip::app::AttributePathParams attributePathParams[1];
+    attributePathParams[0].mEndpointId  = endpointId;
+    attributePathParams[0].mClusterId   = clusterId;
+    attributePathParams[0].mAttributeId = attributeId;
+
+    chip::app::ReadPrepareParams params(device->GetSecureSession().Value());
+    params.mpEventPathParamsList        = nullptr;
+    params.mEventPathParamsListSize     = 0;
+    params.mpAttributePathParamsList    = attributePathParams;
+    params.mAttributePathParamsListSize = 1;
+    params.mMinIntervalFloorSeconds     = minInterval;
+    params.mMaxIntervalCeilingSeconds   = maxInterval;
+    params.mIsFabricFiltered            = fabricFiltered;
+
+    mSubscribeClient =
+        std::make_unique<chip::app::ReadClient>(chip::app::InteractionModelEngine::GetInstance(), device->GetExchangeManager(),
+                                                mBufferedReadAdapter, chip::app::ReadClient::InteractionType::Subscribe);
+    VerifyOrReturnError(mSubscribeClient != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    return mSubscribeClient->SendRequest(params);
+}
+
+void InteractionModel::Shutdown()
+{
+    mReadClient.reset();
+    mSubscribeClient.reset();
+    mWriteClient.reset();
+    mCommandSender.reset();
+}
+
+/////////// ReadClient Callback Interface /////////
+void InteractionModel::OnAttributeData(const chip::app::ConcreteDataAttributePath & path, chip::TLV::TLVReader * data,
+                                       const chip::app::StatusIB & status)
+{
+    OnResponse(status, data);
+}
+
+void InteractionModel::OnEventData(const chip::app::EventHeader & eventHeader, chip::TLV::TLVReader * data,
+                                   const chip::app::StatusIB * status)
+{
+    OnResponse(*status, data);
+}
+
+void InteractionModel::OnError(CHIP_ERROR error)
+{
+    chip::app::StatusIB status(error);
+    OnResponse(status, nullptr);
+}
+
+void InteractionModel::OnDone()
+{
+    mReadClient.reset();
```

Comment:
Why just reset the readclient here?  What about the others?  At least subscribe client?

Reply:
Resetting the subscribe client here would mean the end of the interaction. This is not obvious that this is the expected behaviour at the moment.
For example I have added some commands to `chip-tool` that are meant to be used in "interactive" mode in order to stop the subscription (`examples/chip-tool/commands/clusters/SubscriptionsCommands.h`) 
I think that a YAML command to do the same is needed for tests - so the test can decide the behaviour of a given subscription.

Reply:
Right, but if we are getting `OnDone` that means the read client itself has shut down.  The interaction _has_ ended.  The only question is whether we release the memory.

Reply:
I did have changed it but since InteractionModel is a singleton it ends up closing the subscribe client if a read is done afterward. So I have reverted the change and I have added a TODO. I need to do more work on this code for this change.